src/client.rs

```diff
@@ -27,32 +28,22 @@ struct GraphQLResponse<T> {
   errors: Option<Vec<GraphQLErrorMessage>>,
 }
 
-impl GQLClient {
-  #[cfg(target_arch = "wasm32")]
-  fn client(&self) -> Result<reqwest::Client, GraphQLError> {
-    Ok(Client::new())
-  }
-
-  #[cfg(not(target_arch = "wasm32"))]
-  fn client(&self) -> Result<reqwest::Client, GraphQLError> {
-    Ok(
-      Client::builder()
-        .timeout(std::time::Duration::from_secs(5))
-        .build()
-        .map_err(|e| GraphQLError::with_text(format!("Can not create client: {:?}", e)))?,
-    )
-  }
-}
-
-impl GQLClient {
-  pub fn new(endpoint: impl AsRef<str>) -> Self {
+impl<'a> GQLClient<'a> {
+  pub fn new(endpoint: &'a str) -> Self {
     Self {
-      endpoint: endpoint.as_ref().to_string(),
-      header_map: HeaderMap::new(),
+      endpoint: &endpoint,
+      client: if cfg!(target_arch = "wasm32") {
+        Client::new()
+      } else {
+        Client::builder()
+          .timeout(Duration::from_secs(5))
+          .build()
+          .unwrap()
+      },
     }
   }
 
-  pub fn new_with_headers(endpoint: impl AsRef<str>, headers: HashMap<&str, &str>) -> Self {
+  pub fn new_with_headers(endpoint: &'a str, headers: HashMap<&str, &str>) -> Self {
```

Comment:
Sorry my mistake, I'm missing something here. `header_map` still exists. but there I suggest you still use `impl AsRef<str>`. because do this you can call this function use  &str/String and or support `AsRef<str>` struct